### PR TITLE
[EMB-186] [EMB-240] [EMB-263] registrations tab + improvements

### PR DESCRIPTION
**Pull request by jamescdavis**
## Purpose

Implement node registrations tab, including improvements from [EMB-240](https://openscience.atlassian.net/browse/EMB-240) and [EMB-263](https://openscience.atlassian.net/browse/EMB-263). This PR also implements "skeleton screen" placeholders for node/registration forks as well as linked projects on analytics.

### Added
- Models:
    - `registration-metaschema` (including related adapter & serializer)
- Components:
    - `draft-registrations-blurb` - used for lists of draft registrations
- Routes:
    - `guid-node/registrations` - registrations tab

### Changed
- Models:
    - `osf-model` - add `relatedCounts` attribute and `incrementRelatedCount()`/`decrementRelatedCount()` methods
    - `registration` - add `archiving` attribute and `registrationSchema` relationship
    - `draft-registration` - changed `registrationSchema` relationship type to be `registration-metaschema`
- Adapters:
    - `draft-registration` - override `urlForCreateRecord()` to `POST` to `nodes/{guid}/draft_registrations`
- Serializers:
    - `osf-serializer` - populate `relatedCounts` attribute from relationship meta
- Services:
    - `route-context` - added ability to pass query params to `setGuid()`/`loadModel()`
- Components:
    - `node-blurb` - add `registration` type, optional tags display, and placeholder when `node` is not set
    - `node-navbar` - use `linkTo` for registrations
    - `paginated-relation` - add ability to specify placeholders, and pass actions to items for incrementing/decrementing count
- Routes:
    - `guid-node` - request `forks`, `registrations`, and `draft_registrations` related counts when resolving guid
    - `guid-node/forks` - use placeholder for forks list
    - `guid-registration` - request `forks` related count when resolving guid
    - `guid-registration/forks` - use placeholder for forks list
    - `resolve-guid/resolved-guid-route` - pass-through query params to `routeContext.setGuid()`
- Engines:
    - `analytics-page` - use placeholder for linked nodes list
- Misc:
    - install `@cos-forks/ember-content-placeholders`

### Removed
- Models:
    - `metaschema` (including related adapter & serializer)

## Side Effects / Testing Notes

This PR modifies forks and analytics to use placeholders while loading, so those pages should be lightly tested as well.

## Ticket

https://openscience.atlassian.net/browse/EMB-186
https://openscience.atlassian.net/browse/EMB-240
https://openscience.atlassian.net/browse/EMB-263

# Reviewer Checklist

- [ ] meets requirements
- [ ] easy to understand
- [ ] DRY
- [ ] testable and includes test(s)
- [ ] changes described in `CHANGELOG.md`

### CODE REVIEW

File: lib/osf-components/addon/components/paginated-relation/template.hbs [hunk 1]
```diff
@@ -1,12 +1,22 @@
 {{#if loadItemsTask.isRunning}}
-    {{loading-indicator dark=true}}
+    {{#if (and usePlaceholders placeholderCount)}}
+        <ul class="list-group m-md">
+            {{#each (range 1 placeholderCount) as |item index|}}
+                {{yield (hash
+                    placeholder=(component 'paginated-relation/x-item' item=item index=index)
```
Comment (jamescdavis) [inline lib/osf-components/addon/components/paginated-relation/template.hbs]: Maybe just remove item and index?

File: lib/osf-components/addon/components/paginated-relation/component.ts [hunk 2]
```diff
@@ -73,6 +74,23 @@ export default class PaginatedList extends Component {
...
+    }
+
+    @computed('count')
+    get placeholderCount() {
+        if (this.count && this.count > this.pageSize) {
+            const pages = Math.ceil(this.count / this.pageSize);
+            if (this.page < pages) {
+                return this.pageSize;
+            }
+            return this.count - ((pages - 1) * this.pageSize);
```
Comment (jamescdavis) [inline lib/osf-components/addon/components/paginated-relation/component.ts]: Why no use `%`?

**Comment by brianjgeiger** [comment]
@jamescdavis Is this using the `ember_project_registrations_page` waffle flag? If not, could you add? Also, we should add a "Waffle" section to the PR template, which should include the name of the flag as well as notes for how it should work with the flag off vs. on.

File: app/models/node.ts [hunk 3]
```diff
@@ -121,6 +123,26 @@ export default class Node extends BaseFileItem.extend(Validations) {
      */
     @bool('meta.anonymous') isAnonymous!: boolean;
 
+    /**
+     * Does the current user have write permission on this node?
+     * @property canEdit
+     * @type boolean
+     */
+    @computed('currentUserPermissions')
+    get canEdit() {
```
Comment (jamescdavis) [inline app/models/node.ts]: Let's be more specific, e.g. `currentUserCanEdit`

**Pull request merged**